File: 0053-maximum-subarray/solution.c
int maxSubArray(int* a, int n) {
    int cur = a[0];
    int max = a[0];
    for(int i=1;i<n;i++){
        cur = cur + a[i];
        if(cur<a[i]){
            cur =a[i];
        }
        if(max<cur){
            max=cur;
        }
    }
    return max;
}

File: 0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cur = nums[0];
        int max = nums[0];
        int n = nums.length;
        for(int i = 1;i<n; i++){
            cur += nums[i];
            if(cur < nums[i]) cur = nums[i];
            if(max < cur) max = cur;
        }
        return max;
    }
}
